frame/treasury/src/lib.rs

```diff
@@ -386,10 +393,11 @@ decl_module! {
 		/// # </weight>
 		#[weight = SimpleDispatchInfo::FixedOperational(100_000)]
 		fn approve_proposal(origin, #[compact] proposal_id: ProposalIndex) {
-			T::ApproveOrigin::ensure_origin(origin)?;
+			T::ApproveOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)?;
```

Comment:
I see, using origin to be ApproveOrigin or root instead of just ApproveOrigin allows it be called during benchmark.

Note that this pattern can only works if we don't make any use of the Success type.

(Maybe a broader solution can be found that would also be usable for when Success type is used. because such a broader solution will have to be used anyway for democracy for instance)

Reply:
This was a solution that @gavofyork said would be okay for these situations. Otherwise, we may need to extend traits like I have done for `Contains`. I dunno about any other better solutions :/

But I am very open to ideas!